docs/source/dev/workflow.rst

```diff
@@ -38,32 +38,86 @@ have a GitHub account and that you are able to create/push to repositories on Gi
-Note that a committer may not review their own pull request
-(i.e., an independent code review is required).
+to get a sense of how the branching model works. 
+
+There are two main branches that always exist: ``master`` and ``Developing``. 
+The ``master`` branch is a stable branch that contains the latest release of the 
+code. The ``Developing`` branch is where any ongoing development takes place 
+prior to a release and is not guaranteed to be stable. 
+
+Other branches will be created as required for different purposes which are 
+listed below. When any of these branches need to be merged (e.g., into the 
+``Developing`` branch), a pull request (PR) should be initiated on GitHub that is 
+then reviewed by an approver (see `Requirements for a successful merge`_). If the 
+pull request is satisfactory, it is then merged into develop. Note that a 
+committer may not review their own pull request (i.e., an independent code 
+review is required). 
+
+* **Feature** - All new features, enhancements, and bug fixes should be 
+  developed on a ``Feature`` branch off ``Developing``. Most branches on the 
+  repository will be of this type and consist of work by the core JADE team.
+* **Benchmark** - This type of branch is similar to that above, except it is 
+  specifically for adding new benchmarks. This type of branch will be the most 
+  frequently Pull Requested (PR'd) from forks outside the core team.
+* **Release** - This type of branch will come off ``Developing`` when  the project 
+  leader decides that a release should occur. This is solely for allowing further 
+  testing and bug fixes for an upcoming release and will be merged into ``main`` 
+  once completed.
+* **Hotfix** - This is an exceptional type of branch that comes directly off ``main`` 
+  and is merged directly back into it. This would used when there is a major bug
+  needs fixing immediately.
+
+Release Model
+-------------
+
+The JADE project uses semantic versioning in the format **Major.Minor.Patch** 
+with the choice of versioning increment for a specific release using the 
+following guidelines but the final decision beign discrentionary.
+
+* **Major** - Major or non-backwards compatable functionality additions such as a 
+  new transport code being added, the class structure being modified or the 
+  input/output directory structure is changed.
+* **Minor** - Addition of new features that are additive/backward compatible with 
+  the previous version. New Benchmarks will generally be at this level.
+* **Patch** - Fixes for bugs, typos, tidying code, and other small changes that do
+  not change or extend JADE's intended functionality.
 
 Contribution steps
 ------------------
-These steps apply to both new features and bug fixes. The general steps for contributing
-are as follows:
+JADE follows the development process outlined in the diagram below, relying on 
+Github Issues to track requests and delivery of new features, benchmarks, 
+bugfixes and other developments.
```

Comment:
I think just add another sentence here for emphasis. Something like, 'For example, contributors wishing to add a new benchmark to JADE should first submit an issues using the template for feature requests.'.

Reply:
Added.